pageserver/src/http/openapi_spec.yml

```diff
@@ -93,6 +93,47 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/Error"
+    delete:
+      description: |
+        Attempts to delete specified tenant. 500 and 409 errors should be retried until 404 is retrieved.
+        404 means that deletion successfully finished"
```

Comment:
Hmm.  404 could also mean that someone detached the tenant, right?

I wonder if we should just leave remote deletion markers in place forever (they're cheap) and use that to be able to unambiguously distinguish between a deletion being complete and a tenant just not being attached for some other reason.  It would also be useful if there is some bug in deletion, and we later want to clean up some layer files -- it enables us to be certain that a particular tenant was intentionally deleted.

Reply:
IIRC the scheme this PR is proposing is already used by timeline deletions (even though it's not documented in the very same way as tenant deletions).

Reply:
> Hmm. 404 could also mean that someone detached the tenant, right?

That sounds correct. Though I wont expect that. If tenant is being deleted why detach it in the first place? It wont be used ever again. Detach calls to `tenant.shutdown` which will error out because tenant is already in stopping state. In all remaining cases 404 indeed means successful deletion. 

> I wonder if we should just leave remote deletion markers in place forever (they're cheap)

This was brought up in RFC discussion I think, but IMO we should try to maintain zero garbage policy.
1. It still costs something.
2. It makes investigations trickier (you need to dig through the pile of garbage to get what you need)
3. ListObjects becomes slower and slower if number of entries keeps increasing. We dont do list on `tenants/` during normal operations, but it is useful in manual exploration. This can appear in case we add some background gc-like operation that walks every tenant for stuff to delete like you propose in your RFC